packages/strapi-design-system/src/ProgressBar/ProgressBar.tsx

```diff
@@ -1,11 +1,21 @@
-import React from 'react';
-
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { Box } from '../Box';
+import { Box, BoxProps } from '../Box';
+
+type Enumerate<N extends number, Acc extends number[] = []> = Acc['length'] extends N
+  ? Acc[number]
+  : Enumerate<N, [...Acc, Acc['length']]>;
+
+type IntRange<F extends number, T extends number> = Exclude<Enumerate<T>, Enumerate<F>>;
+
+export interface ProgressBarProps extends Omit<BoxProps, 'background' | 'hasRadius' | 'height' | 'position' | 'width'> {
+  max: IntRange<0, 101>;
+  min: IntRange<0, 101>;
+  size: 'S' | 'M';
+  value: IntRange<0, 101>;
```

Comment:
```suggestion
  max?: IntRange<0, 101>;
  min?: IntRange<0, 101>;
  size?: 'S' | 'M';
  value?: IntRange<0, 101>;
```

You've set defaults below, so these are no longer required